colorbleed/plugins/global/publish/submit_publish_job.py

```diff
@@ -114,6 +182,58 @@ def process(self, instance):
         if not os.path.isdir(output_dir):
             os.makedirs(output_dir)
 
+        if data.get("extendFrames", False):
+
+            family = "colorbleed.imagesequence"
+            override = data["overrideExistingFrame"]
+
+            # override = data.get("overrideExistingFrame", False)
+            out_file = render_job.get("OutFile")
+            if not out_file:
+                raise RuntimeError("OutFile not found in render job!")
+
+            extension = os.path.splitext(out_file[0])[1]
+            _ext = extension[1:]
+
+            # Frame comparison
+            prev_start = None
+            prev_end = None
+            resource_range = range(int(start)-1, int(end)+1)
+
+            # Gather all the subset files (one subset per render pass!)
+            subset_names = [data["subset"]]
+            subset_names.extend(data.get("renderPasses", []))
+
+            for subset_name in subset_names:
+                version = get_latest_version(asset_name=data["asset"],
+                                             subset_name=subset_name,
+                                             family=family)
+
+                # Set prev start / end frames for comparison
+                if not prev_start and not prev_end:
+                    prev_start = version["data"]["startFrame"]
+                    prev_end = version["data"]["endFrame"]
+
+                subset_resources = get_resources(version, _ext)
+                get_resource_files(subset_resources, resource_range, override)
+
+                resources.extend(subset_resources)
+
+            updated_start = start if start < prev_start else prev_start
+            updated_end = end if end > prev_end else prev_end
+
+            # Update metadata and instance start / end frame
+            self.log.info("Updating start / end frame : "
+                          "{} - {}".format(updated_start, updated_end))
+
+            # Start frame
+            metadata["startFrame"] = updated_start
```

Comment:
This overriding of the start and end frame seems right - but it feels off that we're doing this all the way in the Integrator. Logically this should actually be at the point of "collecting" since it's part of knowing how it should be published. But I can see how this is an edge case scenario and is alright to behave a bit oddly - it just feels so weird at this point in time.

Just wanted to have it noted.

Reply:
Yes I agree and it felt weird to place it here but the situation is a bit more complex than just checking ranges.

We need to have the subset's version to retrieve the frame range. 
This means validating if the render pass is an actual subset will need to be prior to the collecting of the render.

The logic is a bit intertwined, I am open to suggestions to improve this. Maybe allow a mediator plugin to handle both the validation and collection.